be/src/storage/object_metastore.cpp

```diff
@@ -0,0 +1,198 @@
+
+    std::string _meta_dir;
+};
+
+Status ObjectMetastore::add_tablet_meta(const TabletMeta& tablet_meta) {
+    auto path = fmt::format("{}/{}{}", _meta_dir, kTabletMeta, tablet_meta.tablet_id());
+    TabletMetaPB tablet_meta_pb;
+    tablet_meta.to_meta_pb(&tablet_meta_pb);
+    auto object = tablet_meta_pb.SerializeAsString();
+    return put_object(path, object);
+}
+
+Status ObjectMetastore::update_tablet_meta(const TabletMeta& tablet_meta) {
+    return add_tablet_meta(tablet_meta);
+}
+
+StatusOr<TabletMetaSharedPtr> ObjectMetastore::get_tablet_meta(TTabletId tablet_id, TSchemaHash schema_hash) {
+    auto path = fmt::format("{}/{}{}", _meta_dir, kTabletMeta, tablet_id);
+    ASSIGN_OR_RETURN(auto meta_binary, get_object(path));
+    auto tablet_meta = std::make_shared<TabletMeta>();
+    RETURN_IF_ERROR(tablet_meta->deserialize(meta_binary));
+    return tablet_meta;
+}
+
+Status ObjectMetastore::remove_tablet_meta(TTabletId tablet_id, TSchemaHash schema_hash) {
+    auto path = fmt::format("{}/{}{}", _meta_dir, kTabletMeta, tablet_id);
+    return delete_object(path);
+}
+
+Status ObjectMetastore::add_rowset_meta(const RowsetMeta& rowset_meta) {
+    auto path = fmt::format("{}/{}{}_{}", _meta_dir, kRowsetMetaPrefix, rowset_meta.tablet_uid().to_string(),
+                            rowset_meta.rowset_id().to_string());
+    RowsetMetaPB rowset_meta_pb;
+    rowset_meta.to_rowset_pb(&rowset_meta_pb);
+    auto object = rowset_meta_pb.SerializeAsString();
+    return put_object(path, object);
+}
+
+Status ObjectMetastore::update_rowset_meta(const RowsetMeta& rowset_meta) {
+    return add_rowset_meta(rowset_meta);
+}
+
+StatusOr<RowsetMetaSharedPtr> ObjectMetastore::get_rowset_meta(const TabletUid& tablet_uid, const RowsetId& rowset_id) {
+    auto path = fmt::format("{}/{}{}_{}", _meta_dir, kRowsetMetaPrefix, tablet_uid.to_string(), rowset_id.to_string());
+    return get_rowset_meta(path);
+}
+
+StatusOr<RowsetMetaSharedPtr> ObjectMetastore::get_rowset_meta(const std::string& path) {
+    ASSIGN_OR_RETURN(auto meta_binary, get_object(path));
+    RowsetMetaPB rowset_meta_pb;
+    if (!rowset_meta_pb.ParseFromString(meta_binary)) {
+        return Status::InternalError("failed to parse RowsetMetaPB from string");
+    }
+    auto rowset_meta = std::make_shared<RowsetMeta>();
+    if (!rowset_meta->init_from_pb(rowset_meta_pb)) {
+        return Status::InternalError("failed to init RowsetMeta from pb");
+    }
+    return rowset_meta;
+}
+
+StatusOr<std::vector<RowsetMetaSharedPtr>> ObjectMetastore::get_rowset_metas(const TabletUid& tablet_uid) {
+    auto object_prefix = fmt::format("{}/{}{}_", _meta_dir, kRowsetMetaPrefix, tablet_uid.to_string());
+    ASSIGN_OR_RETURN(auto objects, list_objects_by_prefix(object_prefix));
+    std::vector<RowsetMetaSharedPtr> rowset_metas;
+    rowset_metas.reserve(objects.size());
+    for (const auto& object : objects) {
+        auto path = fmt::format("{}/{}", _meta_dir, object);
```

Comment:
We can eliminate the string concatenation by making `list_objects_by_prefix` return the full path of each object.